manager/manager_test.go

```diff
@@ -206,22 +221,14 @@ func TestGetContainerInfoV2Failure(t *testing.T) {
 		NumStats: 2,
 	}
 
-	m, _, handlerMap := expectManagerWithContainers(containers, query, t)
+	m, _, _ := expectManagerWithContainers(containers, query, t)
 
 	// Remove /c1 stats
 	err := m.memoryCache.RemoveContainer(statless)
 	if err != nil {
 		t.Fatalf("RemoveContainer failed: %v", err)
 	}
 
-	// Make GetSpec fail on /c2
-	mockErr := fmt.Errorf("intentional GetSpec failure")
-	failingHandler := containertest.NewMockContainerHandler(failing)
-	failingHandler.On("GetSpec").Return(info.ContainerSpec{}, mockErr)
-	failingHandler.On("Exists").Return(true)
-	*handlerMap[failing] = *failingHandler
```

Comment:
`Mock.On(...)` just appends a call to `Mock.ExpectedCalls` (https://github.com/stretchr/testify/blob/master/mock/mock.go#L202). It might be simpler to just clear out ExpectedCalls. Alternatively, you could just modify the call directly, something like:
```go
failingHandler.ExpectedCalls[0].Return(info.ContainerSpec{}, mockErr)
...
```